iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/BusbarSectionXml.java

```diff
@@ -49,16 +57,20 @@ protected BusbarSectionAdder createAdder(VoltageLevel vl) {
     @Override
     protected BusbarSection readRootElementAttributes(BusbarSectionAdder adder, NetworkXmlReaderContext context) {
         int node = XmlUtil.readIntAttribute(context.getReader(), "node");
-        double v = XmlUtil.readOptionalDoubleAttribute(context.getReader(), "v");
-        double angle = XmlUtil.readOptionalDoubleAttribute(context.getReader(), "angle");
         BusbarSection bbs = adder.setNode(node)
                 .add();
-        context.getEndTasks().add(() -> {
-            Bus b = bbs.getTerminal().getBusView().getBus();
-            if (b != null) {
-                b.setV(v).setAngle(angle);
-            }
+
+        IidmXmlUtil.runUntilMaximumVersion(IidmXmlVersion.V_1_0, context, () -> {
+            double v = XmlUtil.readOptionalDoubleAttribute(context.getReader(), "v");
```

Comment:
Use `IidmXmlUtil.readOptionalDoubleAttributeUntilMaximumVersion` to ensure that maximum version 1.0 is asserted when reading this attribute

Reply:
@MioRtia It looks totally equivalent? Maybe the proposition @annetill mades is better because you clearly see that these two attributes are linked.

Reply:
Not really: here if attributes are present in 1.1, they are not read but the import does not notice the file is invalid (unless there is validation after import) which contradicts what we did elsewhere. If we want to throw an exception if the attributes are present for version >= 1.1, a check outside the `runUntilMaximumVersion` method should be done.

Reply:
In IIDM v1.0, if I put a dummy attribute on an equipment, let's say the busbar section, the code will never throw an exception because we never assert that it's a valid/invalid attribute. This is the responsibility of the XSD to assert that the attribute is not valid. But, you're right, it's not totally equivalent to what we did elsewhere.

You cannot simply replace `readOptionDoubleAttribute` by `readOptionalDoubleAttributeUntilMaximumVersion` in the lambda, because the runnable is run only if the version is 1.0. This piece of code should be rewritten. What you propose is (pseudo-java-code):
```
// Read v or throw if attribute is not expected
double v = IidmXmlUtil.readOptionalDoubleAttributeUntilMaximumVersion(IidmXmlVersion.V_1_0, "v");
// Read angle or throw if attribute is not expected
double angle = IidmXmlUtil.readOptionalDoubleAttributeUntilMaximumVersion(IidmXmlVersion.V_1_0, "angle");
if (IidmVersionXml == IidmXmlVersion.V_1_0) {
// if (!Double.isNan(v) || !Double.isNaN(angle)) {
            context.getEndTasks().add(() -> {
                Bus b = bbs.getTerminal().getBusView().getBus();
                if (b != null) {
                    b.setV(v).setAngle(angle);
                }
            });
}
```

Reply:
Yes, we should do something like that to be consistent with other compatibility checks.

Reply:
After some thinking, I think what we should do is:
- assert the version when there is a deletion or an addition of an XML element
- not assert anything where there is a deletion or an addition of an XML attribute

So @annetill you can keep your original code

Reply:
Ok !